### DISCUSSION TITLE
Jest failed to parse a file

### DISCUSSION THREAD

**Discussion by cam-m** [ngrx/platform#3772] [Q&A]
### Which @ngrx/* package(s) are the source of the bug?

component-store, component, data, effects, entity, eslint-plugin, router-store, schematics, store-devtools, store

### Minimal reproduction of the bug/regression with instructions

I'm sure I'm missing something stupid, but trying commit a PR and getting this when I run yarn test:
```
Test suite failed to run

Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.
```

and the error:

```
SyntaxError: Cannot use import statement outside a module

    > 1 | import { EntityAdapter, createEntityAdapter } from '@ngrx/entity';
        | ^
      2 | import { Comparer, IdSelector } from '@ngrx/entity';
      3 |
      4 | import { EntityDispatcherDefaultOptions } from '../dispatchers/entity-dispatcher-default-options';

      at Runtime.createScriptFromCode (../../node_modules/jest-runtime/build/index.js:1796:14)
      at Object.<anonymous> (src/entity-metadata/entity-definition.ts:1:1)
```

This is just one example, I get the same issue thrown on multiple other spec files

It seemed to work fine 2 months ago?

### Minimal reproduction of the bug/regression with instructions

run yarn test

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

NGRX 15.2.1
Windows 10

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

**Comment by timdeschryver** [comment]
Hi, did you already try to clean your node_modules?
Maybe a build first can also do the trick.

**Comment by cam-m** [comment]
Thanks @timdeschryver, yes I did try deleting node_modules a few times, and just tried your suggestion of building first. still getting the same errors. I did commit and push changes and the ci builds and tests appear to be working so it must be something Im missing in my setup. I'm a bit stumped.

**Comment by markostanimirovic** [comment]
> ### Minimal reproduction of the bug/regression with instructions
> run yarn test

Hi @cam-m 👋 

You probably saw in the issue template description that reproduction is required when reporting a bug. Please provide Stackblitz or GitHub repo with a reproduction.

**Comment by cam-m** [comment]
Thanks @markostanimirovic , I'm not sure how i would provide a reproduction since all i've got is a fork of the ngrx/platform repo with some changes for a pull request.  When I push the changes to the PR the CI run tests appear to work and pass fine. They just don't work on my machine.

I will try checking out the repo fresh and see if that fixes it.

**Comment by markostanimirovic** [comment]
Oh, I thought you have this issue in another repo. 😅 

Did you try:

```
rm -rf node_modules/
yarn
yarn test
```

**Comment by markostanimirovic** [comment]
Also, try switching to Node `14.20.0`.

**Comment by cam-m** [comment]
Ha no worries :)

I did try that with no success (although not entirely at the cmd line).

I’ll post back on the result of a clean repo build.

**Comment by cam-m** [comment]
ooh you might be onto something there, I did recently bump up node so will
look at that too

**Comment by markostanimirovic** [comment]
Hi @cam-m,

Did you fix tests by switching to Node `14.20`?

**Comment by cam-m** [comment]
No that didn’t work unfortunately.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by markostanimirovic** [comment]
I'll convert this issue to a Q&A discussion. I use these versions on my machine (Windows 11 OS):

```
Node: v16.16.0
Yarn: v1.22.19
```

**Comment by cam-m** [comment]
After trying numerous node versions (14, 16, 18)  I just took a fresh clone of the repo, and (after one false start due to what appears to be an internal node issue) I got all tests to run and pass.

None the wiser as to what cause the initial issue, but I can move on.

For the record:

node: v16.19.1
yarn: v1.22.19

**Reply by cam-m** [reply]
Nope, now the new repo is broken too.
